### ISSUE TITLE
switch_case_alignment doesn't work with switch expressions

### ISSUE THREAD

**Issue by futuretap** [realm/SwiftLint#5079]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

A switch statement that evaluates to an expression ([SE0380](https://github.com/apple/swift-evolution/blob/main/proposals/0380-if-switch-expressions.md)) throws the error 

`warning:  Switch and Case Statement Alignment Violation: Case statements should vertically align with their enclosing switch statement (switch_case_alignment)`

##### Complete output when running SwiftLint, including the stack trace and command used

see above

### Environment

* SwiftLint version 0.52.3
* CocoaPods
* No configuration file used (default CocoaPods installation)
* Xcode version 15.0 beta 2 (15A5161b)
* Do you have a sample that shows the issue?

```swift
// This triggers a violation:
let color: UIColor = switch state {
case [.highlighted]:
	.gray
case .selected, [.selected, .highlighted]:
	.white
case .disabled:
	.lightGray
default:
	.clear
}
```

**Comment by SimplyDanny** [comment]
Closing as duplicate of #5080.
